### ISSUE TITLE
Jpeg XL build with GCC compiler does not work correctly

### ISSUE THREAD

**Issue by ScopeBurst** [m-ab-s/media-autobuild_suite#1858]
It seems that Jpeg XL doesn't work correctly if GCC is selected as the main compiler in m-ab-s (it compiles, runs, but doesn't encode, on AVX2 system, without additional custom flags), Clang is fine.
It would also be nice to add ssimulacra and butteraugli metrics (they were included with Jpeg XL, although I didn't check the last build), together with DSSIM these are the three most modern and relevant metrics for static images.

Individual repositories with these metrics and their description, but updated versions are included in Jpeg XL:
SSIMULACRA https://github.com/cloudinary/ssimulacra
Butteraugli https://github.com/google/butteraugli

**Comment by lastrosade** [comment]
I second this.

**Comment by wiiaboo** [comment]
> It seems that Jpeg XL doesn't work correctly if GCC is selected as the main compiler in m-ab-s (it compiles, runs, but doesn't encode, on AVX2 system, without additional custom flags), Clang is fine.

Then open an issue on their bug tracker. We're not everyone's bug fixer.

**Comment by Felixkruemel** [comment]
I opened an issue https://gitlab.com/wg1/jpeg-xl/-/issues/62

**Comment by Felixkruemel** [comment]
@wiiaboo Whcih GCC Version do we use by default?

**Comment by wiiaboo** [comment]
Run `gcc -v` on mintty.

**Comment by Felixkruemel** [comment]
A quote of the bug on jpeg-xl:
```

**Comment by ScopeBurst** [comment]
> Then open an issue on their bug tracker. We're not everyone's bug fixer.
Yes, this has already been done, but as long as GCC compatibility has not been added, it is a good idea to write a warning to m-ab-s about correct compilation only in Clang or something like that (otherwise it can cause problems for people, because no errors are shown during compilation, but encoding does not work).

**Comment by wiiaboo** [comment]
It's weird that it doesn't work on the most used C compiler.

I'll just leave this open.
